lib/object.gi

```diff
@@ -384,7 +384,13 @@ InstallMethod( PrintObj,
     true,
     [ HasName ],
     SUM_FLAGS, # override anything specific
-    function ( obj )  Print( Name( obj ) ); end );
+    function( obj )
+    if ISBOUND_GLOBAL( Name( obj ) ) then
+      Print( Name( obj ) );
```

Comment:
I think this should only be triggered if the variable with that name actually contains the object. I.e. like so:
```suggestion
    local name;
    name := Name( obj );
    if ISBOUND_GLOBAL( name ) and IS_IDENTICAL_OBJ( VALUE_GLOBAL( name ), obj ) then
      Print( name );
```